src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -272,7 +273,11 @@ public void downloadZippedFile(Environment env, HttpServletRequest request,
 						folderNameMap = Utils.updateFolderNameMap(
 								folderNameMap, title, folderName);
 						if (folderNameMap.containsKey(folderName)) {
-							folderName = folderNameMap.get(folderName);
+							// if the folder name have / in it then we are not zipping the file with original name instead the folder
+							// name will contain _ in it. As having the / will have cause the zip library creating inner folders
+							if (!(StringUtils.countOccurrencesOf(folderNameMap.get(folderName), "/") > 1)) {
```

Comment:
What happening at this point of code is the `folderName` we are getting is already  renamed with `_` like `2_forward_slash`  but as per the existing workflow code  try to zip it with original folder name like ` 2/forward/slash`   and hence the bug!.